src/hotspot/cpu/ppc/sharedRuntime_ppc.cpp

```diff
@@ -2352,17 +2184,19 @@ nmethod *SharedRuntime::generate_native_wrapper(MacroAssembler *masm,
   }
 
 
-  // Publish thread state
-  // --------------------------------------------------------------------------
+  if (!is_critical_native) {
+    // Publish thread state
+    // --------------------------------------------------------------------------
 
-  // Use that pc we placed in r_return_pc a while back as the current frame anchor.
-  __ set_last_Java_frame(R1_SP, r_return_pc);
+    // Use that pc we placed in r_return_pc a while back as the current frame anchor.
+    __ set_last_Java_frame(R1_SP, r_return_pc);
```

Comment:
These 2 lines need to get moved before if (!is_critical_native) check.
With this change, the 2 tests have passed with " | os.arch=="ppc64" | os.arch=="ppc64le" | os.arch=="s390x"" added.

Reply:
Ok, I moved those lines and added the other platforms for the basic CriticalNative tests that I modified.  Thank you so much for testing this on the other platforms.